ga4gh/datamodel/datasets.py

```diff
@@ -0,0 +1,109 @@
+"""
+Dataset objects
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import os
+import random
+
+import ga4gh.datamodel as datamodel
+import ga4gh.datamodel.variants as variants
+import ga4gh.datamodel.reads as reads
+
+
+class AbstractDataset(datamodel.DatamodelObject):
+    """
+    The base class of datasets containing variants and reads
+    """
+    def __init__(self):
+        self._variantSetIds = []
+        self._variantSetIdMap = {}
+        self._readGroupSetIds = []
+        self._readGroupSetIdMap = {}
+        self._readGroupIds = []
+        self._readGroupIdMap = {}
+
+    def getDirectory(self):
```

Comment:
It would be nice to docstring these methods in the abstract class. It's not blindingly obvious what the IDs and maps are for, so we should explain that through these comments.